src/utils/crypto.ts

```diff
@@ -39,3 +45,28 @@ export const createCryptoContextFor = (keytag: string): CryptoContext => {
     },
   };
 };
+
+/**
+ * Executes the input function injecting an ephemeral crypto context.
+ * An ephemeral crypto context is a context which is bound to a key
+ * that is just created and is deleted after use.
+ *
+ * @param fn The procedure to be executed
+ * @returns The returned value of the input procedure.
+ */
+export const useEphemeralKey = async <R>(
+  fn: (ephemeralContext: CryptoContext) => Promise<R>
+): Promise<R> => {
+  // Use an ephemeral key to be destroyed after use
+  const keytag = `ephemeral-${uuid.v4()}`;
+  await generate(keytag);
+  const ephemeralContext = createCryptoContextFor(keytag);
+  try {
+    const result = await fn(ephemeralContext);
+    await deleteKey(keytag);
```

Comment:
maybe we can put `deleteKey` into finally {} ?

Reply:
I wasn't sure how `finally` works with async code, I suspected some quirks and I was too lazy to investigate 🙃.

It seems to work as expected. However, I fail to test it on a real device (so I just hope iOS/Android js engine likes it as well).

On the happy path, it works like charm https://github.com/pagopa/io-react-native-wallet/pull/51/commits/8c5dfe769953004e77df4f7b0ea44dc1a5416398

@LazyAfternoons @hevelius any thoughts?